drivers/interrupt_controller/intc_arcv2_irq_unit.c

```diff
@@ -218,9 +218,7 @@ static int arc_v2_irq_unit_device_ctrl(const struct device *dev,
 }
 
 SYS_DEVICE_DEFINE("arc_v2_irq_unit", arc_v2_irq_unit_init,
-		  arc_v2_irq_unit_device_ctrl, PRE_KERNEL_1,
-		  CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+		  arc_v2_irq_unit_device_ctrl, PRE_KERNEL_1, 0);
```

Comment:
Not sure, as that's not my area of expertise, but may we use `KERNEL_INIT_PRIORITY_OBJECTS` instead of hard-coded 0 here?

Reply:
I prefer using 0:
```
grep "PRE_KERNEL_1, 0" -R ./
./arch/x86/core/spec_ctrl.c:SYS_INIT(spec_ctrl_init, PRE_KERNEL_1, 0);
./boards/arm/96b_meerkat96/pinmux.c:SYS_INIT(meerakt96_pinmux_init, PRE_KERNEL_1, 0);
./boards/arm/pico_pi_m4/pinmux.c:SYS_INIT(pico_pi_m4_pinmux_init, PRE_KERNEL_1, 0);
./boards/arm/mimx8mm_evk/pinmux.c:SYS_INIT(mimx8mm_evk_pinmux_init, PRE_KERNEL_1, 0);
./boards/arm/colibri_imx7d_m4/pinmux.c:SYS_INIT(colibri_imx7d_m4_pinmux_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1010_evk/pinmux.c:SYS_INIT(mimxrt1010_evk_init, PRE_KERNEL_1, 0);
./boards/arm/sam4e_xpro/pinmux.c:SYS_INIT(sam4e_xpro_init, PRE_KERNEL_1, 0);
./boards/arm/sam_v71_xult/pinmux.c:SYS_INIT(sam_v71_xplained_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1015_evk/pinmux.c:SYS_INIT(mimxrt1015_evk_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1020_evk/pinmux.c:SYS_INIT(mimxrt1020_evk_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1050_evk/pinmux.c:SYS_INIT(mimxrt1050_evk_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1060_evk/pinmux.c:SYS_INIT(mimxrt1060_evk_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1064_evk/pinmux.c:SYS_INIT(mimxrt1064_evk_init, PRE_KERNEL_1, 0);
./boards/arm/mm_swiftio/pinmux.c:SYS_INIT(mm_swiftio_init, PRE_KERNEL_1, 0);
./boards/arm/sam_e70_xplained/pinmux.c:SYS_INIT(sam_e70_xplained_init, PRE_KERNEL_1, 0);
./boards/arm/udoo_neo_full_m4/pinmux.c:SYS_INIT(udoo_neo_full_m4_init, PRE_KERNEL_1, 0);
./boards/arm/warp7_m4/pinmux.c:SYS_INIT(warp7_m4_pinmux_init, PRE_KERNEL_1, 0);
./boards/arm/nrf9160dk_nrf52840/board.c:SYS_INIT(early_init, PRE_KERNEL_1, 0);
./boards/arm/mimxrt1024_evk/pinmux.c:SYS_INIT(mimxrt1024_evk_init, PRE_KERNEL_1, 0);
./boards/posix/nrf52_bsim/bstests_entry.c:SYS_INIT(bst_fake_device_driver_pre2_init, PRE_KERNEL_1, 0);
./soc/arm/arm/beetle/soc.c:SYS_INIT(arm_beetle_init, PRE_KERNEL_1, 0);
./soc/arm/arm/designstart/soc.c:SYS_INIT(arm_designstart_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam/sam3x/soc.c:SYS_INIT(atmel_sam3x_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam/sam4s/soc.c:SYS_INIT(atmel_sam4s_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam/same70/soc.c:SYS_INIT(atmel_same70_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam/sam4e/soc.c:SYS_INIT(atmel_sam4e_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam/sam4l/soc.c:SYS_INIT(atmel_sam4l_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam/samv71/soc.c:SYS_INIT(atmel_samv71_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam0/common/soc_samd2x.c:SYS_INIT(atmel_samd_init, PRE_KERNEL_1, 0);
./soc/arm/atmel_sam0/common/soc_samd5x.c:SYS_INIT(atmel_samd_init, PRE_KERNEL_1, 0);
./soc/arm/cypress/psoc6/soc.c:SYS_INIT(init_cycfg_platform_wraper, PRE_KERNEL_1, 0);
./soc/arm/nordic_nrf/nrf53/soc.c:SYS_INIT(nordicsemi_nrf53_init, PRE_KERNEL_1, 0);
./soc/arm/nordic_nrf/nrf51/soc.c:SYS_INIT(nordicsemi_nrf51_init, PRE_KERNEL_1, 0);
./soc/arm/nordic_nrf/nrf52/soc.c:SYS_INIT(nordicsemi_nrf52_init, PRE_KERNEL_1, 0);
./soc/arm/nordic_nrf/nrf91/soc.c:SYS_INIT(nordicsemi_nrf91_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_imx/mcimx6x_m4/soc.c:SYS_INIT(mcimx6x_m4_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_imx/mcimx7_m4/soc.c:SYS_INIT(nxp_mcimx7_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_imx/rt/soc.c:SYS_INIT(imxrt_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_imx/mimx8mm6_m4/soc.c:SYS_INIT(nxp_mimx8mm6_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_imx/rt6xx/soc.c:SYS_INIT(nxp_rt600_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/k6x/soc.c:SYS_INIT(k6x_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/kl2x/soc.c:SYS_INIT(kl2x_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/kwx/soc_kw2xd.c:SYS_INIT(kw2xd_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/kwx/soc_kw4xz.c:SYS_INIT(kwx_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/k2x/soc.c:SYS_INIT(fsl_frdm_k22f_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/k8x/soc.c:SYS_INIT(k8x_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/ke1xf/soc.c:SYS_INIT(ke1xf_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_kinetis/kv5x/soc.c:SYS_INIT(kv5x_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_lpc/lpc54xxx/soc.c:SYS_INIT(nxp_lpc54114_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_lpc/lpc11u6x/soc.c:SYS_INIT(nxp_lpc11u6x_init, PRE_KERNEL_1, 0);
./soc/arm/nxp_lpc/lpc55xxx/soc.c:SYS_INIT(nxp_lpc55xxx_init, PRE_KERNEL_1, 0);
./soc/arm/silabs_exx32/common/soc.c:SYS_INIT(silabs_exx32_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32f0/soc.c:SYS_INIT(stm32f0_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32f1/soc.c:SYS_INIT(stm32f1_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32f2/soc.c:SYS_INIT(stm32f2_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32f3/soc.c:SYS_INIT(stm32f3_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32f4/soc.c:SYS_INIT(st_stm32f4_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32f7/soc.c:SYS_INIT(st_stm32f7_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32l0/soc.c:SYS_INIT(stm32l0_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32l4/soc.c:SYS_INIT(stm32l4_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32g0/soc.c:SYS_INIT(stm32g0_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32g4/soc.c:SYS_INIT(stm32g4_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32h7/soc_m4.c:SYS_INIT(stm32h7_m4_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32h7/soc_m7.c:SYS_INIT(stm32h7_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32l1/soc.c:SYS_INIT(stm32l1_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32l5/soc.c:SYS_INIT(stm32l5_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32mp1/soc.c:SYS_INIT(stm32m4_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32wb/soc.c:SYS_INIT(stm32wb_init, PRE_KERNEL_1, 0);
./soc/arm/st_stm32/stm32wl/soc.c:SYS_INIT(stm32wl_init, PRE_KERNEL_1, 0);
./soc/arm/ti_simplelink/cc32xx/soc.c:SYS_INIT(ti_cc32xx_init, PRE_KERNEL_1, 0);
./soc/arm/ti_simplelink/msp432p4xx/soc.c:SYS_INIT(ti_msp432p4xx_init, PRE_KERNEL_1, 0);
./soc/arm/ti_simplelink/cc13x2_cc26x2/soc.c:SYS_INIT(ti_cc13x2_cc26x2_init, PRE_KERNEL_1, 0);
./soc/arm/bcm_vk/valkyrie/soc.c:SYS_INIT(valkyrie_init, PRE_KERNEL_1, 0);
./soc/arm/bcm_vk/viper/soc.c:SYS_INIT(viper_init, PRE_KERNEL_1, 0);
./soc/arm/quicklogic_eos_s3/soc.c:SYS_INIT(eos_s3_init, PRE_KERNEL_1, 0);
./soc/riscv/openisa_rv32m1/soc.c:SYS_INIT(soc_rv32m1_init, PRE_KERNEL_1, 0);
./soc/arm64/bcm_vk/viper/soc.c:SYS_INIT(viper_init, PRE_KERNEL_1, 0);
./soc/arm64/nxp_layerscape/ls1046a/soc.c:SYS_INIT(ls1046a_init, PRE_KERNEL_1, 0);
./tests/kernel/device/src/main.c:SYS_INIT(pre1_fn, PRE_KERNEL_1, 0);
......
```

Reply:
I'm fine with that as well. Anyways, we'll let more knowledgeable people to decide ;)